pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -301,16 +301,14 @@ private Op convertOp(MetadataOp op) {
 
         try {
             zkc.exists(path, null, (rc, path1, ctx, stat) -> {
-                execute(() -> {
```

Comment:
Use the execute is unnecessary, the executor is only used to execute flush, or add the request to the queue in zk callback.

Reply:
I don't think it's unnecessary here, otherwise we're triggering the future from the ZK event thread.